lib/dynamoid/criteria/chain.rb

```diff
@@ -136,27 +140,41 @@ def each(&block)
       #
       # @since 0.2.0
       def records
+        pages.lazy.flat_map { |i| i }
```

Comment:
the explicit lazy turns out to be needed here, or the flat_map ended up calling scan/query